pi-apps-terminal-bash-edition.sh

```diff
@@ -148,106 +90,154 @@ while [ "$1" != "" ]; do
+				exit $?
+			fi
 		;;
-		multi-install)
-			shift
-			for arg in "$@"; do
-				cmdflags+="$arg\n"
-			done
-			#remove last \'n'
-			args=${cmdflags%\\n}
-			#install apps
-			$PI_APPS_DIR/manage multi-install "$(echo -e "$args")"
-			exit $?
-			;;
 		remove|uninstall)
 			shift
-			for arg in "$@"; do
-				cmdflags+="$arg "
-			done
-			$PI_APPS_DIR/manage uninstall "$(echo $cmdflags)"
-			exit $?
+			if [ -d "$DIRECTORY/apps/$*" ]; then #if only provides one app
+				for arg in "$@"; do
+					cmdflags+="$arg "
+				done
+				"$DIRECTORY/manage" uninstall "$cmdflags"
+				exit $?
+			else #multiple app
+				for arg in "$@"; do
+					cmdflags+="$arg\n"
+				done
+				#remove last \'n'
+				args=${cmdflags%\\n}
+				#uninstall apps
+				"$DIRECTORY/manage" multi-uninstall "$(echo -e "$args")"
+				exit $?
+			fi
 		;;
-		multi-remove | multi-uninstall)
-			shift
-			for arg in "$@"; do
-				cmdflags+="$arg\n"
-			done
-			args=${cmdflags%\\n}
-			#uninstall apps
-			$PI_APPS_DIR/manage multi-uninstall "$(echo -e "$args")"
-			exit $?
-			;;
 		reinstall)
 			shift
-			for arg in "$@"; do
-				cmdflags+="$arg "
-			done
-			cmdflags="${cmdflags::-1}"
-			$PI_APPS_DIR/manage uninstall "$cmdflags"
-			$PI_APPS_DIR/manage install "$cmdflags" || error "Failed to reinstall \"$cmdflags\"!"
-			exit $?
+			if [ -d "$DIRECTORY/apps/$*" ]; then #if only provides one app
+				for arg in "$@"; do
+					cmdflags+="$arg "
+				done
+				"$DIRECTORY/manage" uninstall "$cmdflags"
+				"$DIRECTORY/manage" install "$cmdflags"
+				exit $?
+			else #multiple app
+				for arg in "$@"; do
+					cmdflags+="$arg\n"
+				done
+				#remove last \'n'
+				args=${cmdflags%\\n}
+				"$DIRECTORY/manage" multi-uninstall "$(echo -e "$args")"
+				"$DIRECTORY/manage" multi-install "$(echo -e "$args")"
+				exit $?
+			fi
 		;;
 		list-installed)
 			#list all the installed apps
-			#list_apps installed
-			ls "$PI_APPS_DIR/apps" | GREP_COLORS='ms=1;34' grep --color=always -x "$(grep -rx 'installed' "${PI_APPS_DIR}/data/status" | awk -F: '{print $1}' | sed 's!.*/!!' | sed -z 's/\n/\\|/g' | sed -z 's/\\|$/\n/g')"
+			list_apps installed
 			exit $?
 		;;
 		list-uninstalled)
 			#list all the uninstalled apps
-			#list_apps uninstalled
-			ls $PI_APPS_DIR | grep --color=always -x "$(grep -rx 'uninstalled' "${PI_APPS_DIR}/data/status" | awk -F: '{print $1}' | sed 's!.*/!!' | sed -z 's/\n/\\|/g' | sed -z 's/\\|$/\n/g')"
-			ls $PI_APPS_DIR | grep --color=always -vx "$(ls "${PI_APPS_DIR}/data/status" | sed -z 's/\n/\\|/g' | sed -z 's/\\|$/\n/g')"
+			list_apps uninstalled
 			exit $?
 		;;
 		list-corrupted)
 			#list all the corrupted apps
-			#list_apps corrupted
-			ls $PI_APPS_DIR/apps | grep --color=always -x "$(grep -rx 'corrupted' "${PI_APPS_DIR}/data/status" | awk -F: '{print $1}' | sed 's!.*/!!' | sed -z 's/\n/\\|/g' | sed -z 's/\\|$/\n/g')"
+			list_apps corrupted
 			exit $?
 		;;
 		list-all)
 			#list all the apps
-			list-all
+			list_apps cpu_installable
```

Comment:
I don't want this change, it doesn't print the description of the apps, and doesn't format the output.
Could you instead update my `list-all` function to list only available apps please?
something like this would probably work (not tested):
```bash
for app in $(list_apps cpu_installable); do
    echo -e "\n${bold}${inverted}${light_blue}$app${normal}"
    echo -e "${green}$(cat "$DIRECTORY/apps/$app")${normal}"
done
```

Reply:
This works for me: 

```bash
IFS=$'\n'
for app in $(list_apps cpu_installable); do
     echo -e "\n${bold}${inverted}${light_blue}$app${normal}"
     echo -e "${green}$(cat "$DIRECTORY/apps/$app/description")${normal}"
done
```

But I don't think all users need the descriptions, so I made a `-d` flag to show descriptions, else it will only show the app list, without descriptions. 

https://github.com/cycool29/PiApps-terminal_bash-edition/commit/a4ead87b5c2b7bfd838aba038817574a1ac0896d